lib/common/socket.c

```diff
@@ -1437,6 +1437,42 @@ void h2o_ssl_register_npn_protocols(SSL_CTX *ctx, const char *protocols)
 
 #endif
 
+int h2o_socket_set_df_bit(int fd, int domain)
+{
+#define SETSOCKOPT(ip, optname, _optvar)                                                                                           \
+    do {                                                                                                                           \
+        int optvar = _optvar;                                                                                                      \
+        if (setsockopt(fd, ip, optname, &optvar, sizeof(optvar)) != 0) {                                                           \
+            perror("setsockopt(" H2O_TO_STR(optname) ")");                                                                         \
+            return 0;                                                                                                              \
+        }                                                                                                                          \
+        return 1;                                                                                                                  \
+    } while (0)
+
+    switch (domain) {
+    case AF_INET:
+#if defined(IP_PMTUDISC_DO)
+        SETSOCKOPT(IPPROTO_IP, IP_MTU_DISCOVER, IP_PMTUDISC_DO);
+#elif defined(IP_DONTFRAG)
+        SETSOCKOPT(IPPROTO_IP, IP_DONTFRAG, 1);
+#endif
+        break;
+    case AF_INET6:
+#if defined(IPV6_PMTUDISC_DO)
+        SETSOCKOPT(IPPROTO_IPV6, IPV6_MTU_DISCOVER, IPV6_PMTUDISC_DO);
+#elif defined(IPV6_DONTFRAG)
+        SETSOCKOPT(IPPROTO_IPV6, IPV6_DONTFRAG, 1);
+#endif
+        break;
+    default:
+        break;
+    }
+
+    return 1;
```

Comment:
So the intention is that "if the underlying system doesn't support `IP_DONTFRAG`, that's fine, we'll ignore it and move on without DF." I guess that's fine because 1) setting DF is ultimately an optimization, 2) probably most operating systems we mainly target support `IP_DONTFRAG`? Just wanted to make sure this is an intended behavior.

Reply:
Thank you for the review. That's a keep observation, and you are correct.

I've added clarifications to the doc comment, please see ec46562.